src/ray/raylet/node_manager.cc

```diff
@@ -1889,6 +1899,20 @@ void NodeManager::HandleCancelResourceReserve(
   RAY_LOG(DEBUG) << "Request to cancel reserved resource is received, "
                  << bundle_spec.DebugString();
 
+  // Cancel lease requests that are waiting for workers
+  // to free the acquired pg bundle resources
+  // so that pg bundle can be returned.
+  local_task_manager_->CancelTasks(
+      [&](const std::shared_ptr<internal::Work> &work) {
+        const auto bundle_id = work->task.GetTaskSpecification().PlacementGroupBundleId();
+        return (bundle_id.first == bundle_spec.PlacementGroupId()) &&
+               (work->GetState() == internal::WorkStatus::WAITING_FOR_WORKER);
```

Comment:
Fix for the revert: CancelResourceReserve rpc might be called when prepare failed and in this case, we shouldn't cancel the lease request since the PG will be retried and created eventually. So this fix only limits the cancellation to the lease requests that are waiting for workers since this is the case where we need to free bundle resources.